website/frontend/views.py

```diff
@@ -308,7 +308,13 @@ def swap_http_https(url):
                          ("http:", "https:")]:
         if url.startswith(one):
             return other+url[len(one):]
-    raise ValueError("URL doesn't start with http or https")
+    raise ValueError("URL doesn't start with http: or https: ({})".format(url))
+
+def decode_scheme_colon(url):
+    # Sometimes the colon of http: or https: is URL-encoded.
+    # Sometimes the encoding percent sign is itself encoded (multiple times!)
+    # So replace as many %25 as necessary to get to the colon
+    return re.sub('http(s?)%(25)*3A', 'http\g<1>:', url)
```

Comment:
Why would this happen?  I don't think we should have to guess what the encoding scheme is.

Reply:
This was from some errors that Jenny forwarded to me.